core/src/main/java/nl/nn/adapterframework/http/rest/ApiServiceDispatcher.java

```diff
@@ -171,27 +175,34 @@ protected JsonObject generateOpenApiJsonSchema(Collection<ApiDispatchConfig> cli
 
 		for (ApiDispatchConfig config : clients) {
 			JsonObjectBuilder methods = Json.createObjectBuilder();
+			ApiListener listener = null;
 			for (String method : config.getMethods()) {
 				JsonObjectBuilder methodBuilder = Json.createObjectBuilder();
-				ApiListener listener = config.getApiListener(method);
+				listener = config.getApiListener(method);
 				if(listener != null && listener.getReceiver() != null) {
 					IAdapter adapter = listener.getReceiver().getAdapter();
 					if (StringUtils.isNotEmpty(adapter.getDescription())) {
 						methodBuilder.add("summary", adapter.getDescription());
 					}
-					methodBuilder.add("operationId", adapter.getName());
+					if(StringUtils.isNotEmpty(listener.getOperationId())) {
+						methodBuilder.add("operationId", listener.getOperationId());
+					}
 					// GET and DELETE methods cannot have a requestBody according to the specs.
 					if(!method.equals("GET") && !method.equals("DELETE")) {
 						mapRequest(adapter, listener.getConsumesEnum(), methodBuilder);
 					}
+					mapParamsInRequest(adapter, listener, methodBuilder);
+
 					//ContentType may have more parameters such as charset and formdata-boundry
 					MediaTypes produces = listener.getProducesEnum();
 					methodBuilder.add("responses", mapResponses(adapter, produces, schemas));
 				}
-
 				methods.add(method.toLowerCase(), methodBuilder);
 			}
-			paths.add("/"+config.getUriPattern(), methods);
+			if(listener != null) {
+				String uriPattern = listener.getUriPattern().startsWith("/") ? listener.getUriPattern() : "/" + listener.getUriPattern();
```

Comment:
I think the `getUriPattern()` should always return a usable pattern with or without prepending slash. 
That way this 'check' would not be necessarily, and we will have a uniform way to displaying the pattern.